python/pyspark/pandas/tests/test_typedef.py

```diff
@@ -56,6 +56,23 @@
 
 
 class TypeHintTests(unittest.TestCase):
+    def test_infer_schema_with_no_return(self):
+        def try_infer_return_type():
+            def f():
+                pass
+
+            infer_return_type(f)
+
+        self.assertRaisesRegex(TypeError, "Unsupported callable", try_infer_return_type)
+
+        def try_infer_return_type():
+            def f() -> None:
+                pass
+
+            infer_return_type(f)
+
+        self.assertRaisesRegex(TypeError, "not understood", try_infer_return_type)
```

Comment:
This too, It would be great if we can show the message more explicitly.

e.g. "None type is not acceptable for the function passed to `infer_return_type`."

or something.